draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1281,19 +1290,18 @@ ciphersuite, respectively, and \<suiteID\> is the Suite ID of the
 encoding used in ciphersuite \<yy\>.
 
 As another example, consider a fictional protocol named Baz that requires
-two independent random oracles, where one oracle outputs points on the curve
-E1 and the other outputs points on the curve E2.
-Reasonable choices of tags for the E1 and E2 oracles are
-"BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-E1" and "BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-E2",
+two independent random oracles to the same curve.
+Reasonable choices of tags for these oracles are
+"BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-ENC1" and "BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-ENC2",
 respectively, where \<xx\>, \<yy\>, and \<suiteID\> are as described above.
 
 # Utility functions {#utility}
 
 Algorithms in this document make use of utility functions described below.
 
-For security reasons, all field operations, comparisons, and assignments
-MUST be implemented in constant time (i.e., execution time MUST NOT depend
-on the values of the inputs), and without branching.
+For constant-time implementations, all field operations, comparisons, and
```

Comment:
Reading this piece out of context, I wonder if someone might conclude, "oh, so if I don't want constant time, I can ignore these steps." The previous text made it clear that this is necessary "for security reasons." Maybe we can bring that back by saying, "Each function MUST be implemented in constant time. To do so, all field operations..."?

Reply:
Good point. I was intending to flag this edit as something we should discuss:

I'm not convinced it makes sense to say that implementations MUST be constant time, for a few reasons. First, it's not always the case that protocols require constant-time implementations, which I think implies SHOULD rather than must. Second, it's not obvious what it means to require something like this---do we mean that one is just not allowed to implement hash-to-curve in (say) Python, where it's basically impossible to guarantee constant time? (In this case, our reference implementations aren't technically conforming!)

I'd be glad to make this a SHOULD rather than a MUST.

Reply:
Related to the above: in other places in the document it seems like we're more ambivalent about SHOULD vs MUST for constant time. So if we decide that this will be a MUST rather than a SHOULD, we should (must? :smile:) do a sweep for consistency.

Reply:
See also the very first sentence of security considerations:

    When constant-time implementations are required, all basic operations and
    utility functions must be implemented in constant time, as discussed in
    {{utility}}.

Reply:
Thinking about this more: it seems like it might make sense to add or expand a paragraph in the Security Considerations section to address constant timeness.

Reply:
Just pushed a slightly expanded discussion.

Reply:
The new text works for me!